### PR TITLE
TF conv func substitutions

### PR DESCRIPTION
**Pull request by elad-c**
## Pull Request Description:
This PR adds a substitution for replacing tf.nn.conv2d \ depthwise_conv2d \ convolution (possibly with subsquent tf.add \ tf.nn.bias_add) with a Keras Conv2D or DepthwiseConv2D layer.

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+    """
+    b = None
+    next_nodes = _graph.get_next_nodes(_node)
+    if len(next_nodes) == 1 and len(_graph.get_prev_nodes(next_nodes[0])) == 1:
+        # Found pattern in graph: conv_node->next_node. Check if next node is add\bias_add that can be absorbed as bias.
+        if next_nodes[0].is_match_type(tf.add):
+            b = next_nodes[0].weights.get(0, next_nodes[0].weights.get(1))
+            if b is not None and len(b.shape) != 1:
+                # Add node constant shape not regular. Expecting a 1-D array.
```
Comment (eladc-git) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py:54]: Is this add with const case? if so, please add a comment for that.

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            k = np.array(conv_func_node.op_call_kwargs[FILTERS], dtype=np.float32)
+        else:
+            # Conv weight isn't a constant -> skip substitution.
+            return graph  # pragma: no cover
+
+        if len(k.shape) != 4:
+            # Conv dimension doesn't match conv2d dimension (K1 x K2 x Cin x Cout) -> skip substitution.
+            return graph  # pragma: no cover
+
+        # Check is can extract bias from next node.
```
Comment (eladc-git) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py:130]: if

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        b = extract_bias_node_data(conv_func_node, graph)
+
+        weights = {KERNEL: k}
+        # Create Conv2D layer attributes.
+        conv_fw_attr = {FILTERS: k.shape[3], KERNEL_SIZE: k.shape[:2]}
+        if len(conv_func_node.op_call_args) > 0:
+            Logger.critical(f"node {conv_func_node.name} expected to have only kwargs but got args={conv_func_node.op_call_args}.")
```
Comment (eladc-git) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py:137]: add no cover

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+        k = dwconv_func_node.weights[1]
+
+        # Check is can extract bias from next node.
+        b = extract_bias_node_data(dwconv_func_node, graph)
+
+        weights = {DEPTHWISE_KERNEL: k}
+        k_shape = k.shape
+        conv_fw_attr = {DEPTH_MULTIPLIER: k_shape[3], KERNEL_SIZE: k_shape[:2]}
+        if len(dwconv_func_node.op_call_args) > 0:
+            Logger.critical(f"node {dwconv_func_node.name} expected to have only kwargs but got args={dwconv_func_node.op_call_args}.")
```
Comment (eladc-git) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py:214]: no cover

File: tests/keras_tests/feature_networks_tests/test_features_runner.py [hunk 5]
```diff
@@ -805,6 +806,9 @@ def test_keras_tpcs(self):
         TpcTest(f'{C.TFLITE_TP_MODEL}.v1', self).run_test()
         TpcTest(f'{C.QNNPACK_TP_MODEL}.v1', self).run_test()
 
+    def test_conv_func_substitutions(self):
+        ConvSubstitutionsTest(self).run_test()
```
Comment (eladc-git) [inline tests/keras_tests/feature_networks_tests/test_features_runner.py:810]: ConvFuncSubstitutionsTest

**Review by eladc-git — COMMENTED** [review]
small things

**Review by eladc-git — APPROVED** [review]
